upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -1160,6 +1160,46 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
       csi handoff bss-update-param --set "${bootparameter}"
     done
 
+    # Get a list of NCNs.
+    if IFS=$'\n' read -rd '' -a NCN_XNAMES; then
+      :
+    fi <<< "$(cray hsm state components list --role Management --type Node --format json | jq -r '.Components | map(.ID) | join("\n")')"
+
+    # If no NCNs are found we should exit, otherwise if forces its way forward then NCNs will be missing critical packages.
+    if [ "${#NCN_XNAMES[@]}" -eq '0' ]; then
+      echo >&2 'No NCN xnames were found in HSM! Aborting.'
+      exit 1
+    fi
+
+    params=""
+    error=0
+
+    # Loop through one at a time. If `--hosts` isn't provided, we will error out on the 'Global' key.
+    for ncn_xname in "${NCN_XNAMES[@]}"; do
+      printf "% -15s: " "${ncn_xname}"
+
+      params=$(cray bss bootparameters list --hosts "${ncn_xname}" --format json | jq '.[] |."params"' \
+        | sed -E \
+          -e 's/ip=hsn[0-9]+:auto6//g' \
+          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s+//' \
```

Comment:
Two issues.

1. Remove the trailing white space when it exists (when this isn't the final parameter in the list). 1183 was not trimming whitespace, and 1184 was greedily trimming white space. Use `\s?`.
2. The new regex only replaces the first `ifname=hsn`, it'll miss one if two are present (hsn0 and hsn1). Add the `/g` (global) flag in the new regex
```suggestion
          -e 's/ip=hsn[0-9]+:auto6\s?//g' \
          -e 's/ifname=hsn[0-9]+:[0-9a-fA-F:]{17}\s?//g' \
```